core/runtime.rs

```diff
@@ -1218,13 +1215,11 @@ impl JsRuntime {
       }
     }
 
-    // Ops
+    // Resolve async ops, run all next tick callbacks and macrotasks callbacks
+    // and only then check for any promise exceptions (`unhandledrejection`
+    // handlers are run in macrotasks callbacks so we need to let them run
+    // first).
     self.resolve_async_ops(cx)?;
```

Comment:
I should probably rename this function since it also is responsible to responding to JavaScript